src/main/java/operations/AddOperation.java

```diff
@@ -0,0 +1,66 @@
+package operations;
+
+import exceptions.DukeException;
+import tasks.Task;
+import tasks.TaskFactory;
+import tasks.TaskList;
+
+
+/**
+ * AddOperation handle adding command into tasklist
+ */
+public class AddOperation extends Operation {
+
+    /**
+     * Initializes operation with operationName
+     *
+     * @param operationNameLocal Name of the Operation
+     * @param order Order given by User
+     */
+    public AddOperation(String operationNameLocal, String order) throws DukeException {
+        super(operationNameLocal, order);
+    }
+
+    /**
+     * Makes a new Task for add operation
+     * @param taskDescription
+     * @return The new task
+     */
+    protected Task makeTask(String taskDescription) throws DukeException {
+        TaskFactory taskFactory = new TaskFactory("", taskDescription);
+        try {
+            Task newTask = taskFactory.makeTask();
+            return newTask;
+        } catch (DukeException e) {
+            throw e;
+        }
+
+    }
+
+    /**
+     * Adds a new task to tasklist
+     *
+     * @return The report of the operation result
+     */
+    @Override
+    public String operate() throws DukeException {
+        Task newTask;
+        try {
+            newTask = makeTask(order);
+        } catch (DukeException e) {
+            throw e;
+        }
+        if (newTask == null) {
+            return "failed to add";
```

Comment:
Consider handling this with an exception instead, rather than returning a particular value. Rationales are: 1. Better practice, and 2. the caller knows when something wrong occurred.

Also, same issue as Line 62.